markdown/opentelemetry-intro/opentelemetry-intro.md

```diff
@@ -0,0 +1,320 @@
+
+Click on the previously created service and navigating to the **Connect** tab. 
+Expand the "Solace Messaging" menu to get the connection details
+![connection-service](img/connection-service.png "connection-service")
+
+Use a text editor to save the the connection parameters with the **Host**, **Message VPN Name**, **Client Username** and **Password**. You'll need them later.
+
+✔️ You've created the first part of the solution!
+![check1](img/check1.png)
+
+## Set up Jaeger All-In-One
+
+Duration: 0:05:00
+
+Jaeger collects and visualizes the spans created by your microservices.  It has several different components such as an underlying database, a GUI and a span collector, but fortunately it has an all-in-one distribution that makes setup easy.
+
+Negative
+: If you don't have a tool that opens .gz files, you can download one. [7-Zip](https://www.7-zip.org/download.html) is a good free option
+
+### Steps to install
+1. Extract the Jaeger distribution file you downloaded as part of the prerequisites.  (e.g. `jaeger-1.17.1-windows-amd64.tar.gz` on Windows)
+1. Open a command terminal.
+1. Run `jaeger-all-in-one`
+1. Using a web browser, navigate to `http://localhost:16686/`
+1. You should be greeted by a picture of a...chipmunk?
+
+Negative
+: You'll need to keep the terminal window running Jaeger open.  If you close it, you'll lose the traces you've collected.
+
+✔️ You've created the second part of the solution!
+![check2](img/check2.png)
+## Set up Python environment
+
+Duration: 0:12:00
+
+Python is a great language for prototyping applications.  It's easy to get up and running quickly, and since it doesn't compile you just change a line and reload.  This CodeLab uses Python to simulate Salesforce publishing the message and two event consumers using the Solace PubSub+ Python API.  And since there is a native OpenTelemetry API for Python it can produce the OpenTelemetry tracing as well.  
+
+### Steps to install
+1. Open the Python installer that you downloaded in the prerequisites.
+1. You'll be guided through the install process. Be sure to include Python in your PATH variable.
+
+Once it's complete, you should have a program called IDLE (Integrated Development and Learning Environment).
+![IDLE](img/idle.png "IDLE")
+  That's what you'll use to modify and run the Python scripts.
+## Create the "Salesforce" message publisher
+
+Duration: 0:10:00
+
+Positive
+: This CodeLab focuses on OpenTelemetry, so it won't actually implement the Salesforce events.  If that's your passion, Solace has a great [CodeLab about event-enabling Salesforce with Boomi and Solace](https://codelabs.solace.dev/codelabs/boomi-salesforce-codelab/).  Likewise, this CodeLab doesn't get into the nitty gritty of the Python API.  If you want to now more about those capabilities, there will be a CodeLab soon.
+
+1. If you haven't already, unzip the CodeLabs code into a convenient directory
+1. Open a terminal window 
+1. Install the OpenTelemetry APIs with pip:
```

Comment:
Here is a python tip that I usually find helpful tho up to implement. Instead of asking users to install every python library via pip install, I would have a `requirements.txt` file that has 
```
opentelemetry-api==0.12b0
opentelemetry-sdk==0.12b0
```
in it and just prompt the user to `pip install -r requirements.txt`. That way you can then later add the solace python API in the requirements.txt file without needing to update the codelab should there be a need to have the solace API (or other libs) downloaded.

Reply:
Cool!  I will get that going.